### ISSUE TITLE
Allow for a maximum number of concatenations under `prefer-template` rule

### ISSUE THREAD

**Issue by blake-regalia** [eslint/eslint#10017]
**What rule do you want to change?**
`prefer-template`

**Does this change cause the rule to produce more or fewer warnings?**
Fewer.

**How will the change be implemented? (New option, new default behavior, etc.)?**
New object option: `"maxConcat": integer`

**Please provide some example code that this change will affect:**

```js
var str = line + '\n';
```

**What does the rule currently do for this code?**
Prefers template literal.

**What will the rule do after it's changed?**
Allow string concatenation instead of template literals as long as the number of concatenation operations does not exceed the `"maxConcat"` option.

**Justification**
The template literal equivalent for concatenating to one side of a variable is longer and more complex than the former:
```js
var str = `${line}\n`;
// versus
var str = line+'\n';
```

**Comment by not-an-aardvark** [comment]
Thanks for the proposal. I have roughly the same feelings about this option as https://github.com/eslint/eslint/issues/10015#issuecomment-368261571.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
